modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/gnirs/GnirsController.scala

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package edu.gemini.seqexec.server.gnirs
+
+import edu.gemini.seqexec.model.dhs.ImageFileId
+import edu.gemini.seqexec.server.{ObserveCommand, SeqAction}
+import squants.Time
+
+trait GnirsController {
+
+  def observe(obsid: ImageFileId, expTime: Time): SeqAction[ObserveCommand.Result]
+
+  // endObserve is to notify the completion of the observation, not to cause its end.
+  def endObserve: SeqAction[Unit]
+
+  def stopObserve: SeqAction[Unit]
+
+  def abortObserve: SeqAction[Unit]
+
+}
+
+object GnirsController {
+
+  sealed trait Mode
```

Comment:
After analysing the old Seqexec code I think I can merge the acquisitionMirror and crossDispersed parameters into one. This will help managing interdependencies between those parameters. For example, crossDispersed settings don't have any effect if acquisitionMirror is In.